new-new-packages/compiler/src/compiler/rcst.rs

```diff
@@ -0,0 +1,357 @@
+use std::fmt::{self, Display, Formatter};
+
+#[derive(Debug, PartialEq, Eq, Clone, Hash)]
+pub enum Rcst {
+    EqualsSign,         // =
+    Comma,              // ,
+    Colon,              // :
+    OpeningParenthesis, // (
+    ClosingParenthesis, // )
+    OpeningBracket,     // [
+    ClosingBracket,     // ]
+    OpeningCurlyBrace,  // {
+    ClosingCurlyBrace,  // }
+    Arrow,              // ->
+    DoubleQuote,        // "
+    Octothorpe,         // #
+    Whitespace(String), // contains only non-multiline whitespace
+    Newline(String), // the associated `String` because some systems (such as Windows) have weird newlines
+    Comment {
+        octothorpe: Box<Rcst>,
+        comment: String,
+    },
+    TrailingWhitespace {
+        child: Box<Rcst>,
+        whitespace: Vec<Rcst>,
+    },
+    Identifier(String),
+    Symbol(String),
+    Int(u64),
+    Text {
+        opening_quote: Box<Rcst>,
+        parts: Vec<Rcst>,
+        closing_quote: Box<Rcst>,
+    },
+    TextPart(String),
+    Parenthesized {
+        opening_parenthesis: Box<Rcst>,
+        inner: Box<Rcst>,
+        closing_parenthesis: Box<Rcst>,
+    },
+    Call {
+        name: Box<Rcst>,
+        arguments: Vec<Rcst>,
+    },
+    Struct {
+        opening_bracket: Box<Rcst>,
+        fields: Vec<Rcst>,
+        closing_bracket: Box<Rcst>,
+    },
+    StructField {
+        key: Box<Rcst>,
+        colon: Box<Rcst>,
+        value: Box<Rcst>,
+        comma: Option<Box<Rcst>>,
+    },
+    Lambda {
+        opening_curly_brace: Box<Rcst>,
+        parameters_and_arrow: Option<(Vec<Rcst>, Box<Rcst>)>,
+        body: Vec<Rcst>,
+        closing_curly_brace: Box<Rcst>,
+    },
+    Assignment {
+        name: Box<Rcst>,
+        parameters: Vec<Rcst>,
+        equals_sign: Box<Rcst>,
+        body: Vec<Rcst>,
+    },
+    Error {
+        unparsable_input: String,
+        error: RcstError,
+    },
+}
+
+#[derive(Debug, PartialEq, Eq, Clone, Hash)]
+pub enum RcstError {
+    IdentifierContainsNonAlphanumericAscii,
+    SymbolContainsNonAlphanumericAscii,
+    IntContainsNonDigits,
+    TextDoesNotEndUntilInputEnds,
```

Comment:
To unify this with `StructNotClosed` and others:

```suggestion
    TextNotClosed,
```